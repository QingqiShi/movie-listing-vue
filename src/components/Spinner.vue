<template functional>
  <div class="spinner">
    <div class="spinner__wrapper">
      <div class="spinner__A"></div>
      <div class="spinner__B"></div>
      <div class="spinner__C"></div>
    </div>
  </div>
</template>

<script>
/**
 * Purely functional component to display a loading spinner
 */
export default {};
</script>

<style lang="scss">
.spinner {
  &__wrapper {
    width: 3rem;
    height: 3rem;
    display: flex;
    justify-content: space-between;
    margin: 2rem auto;
  }

  &__A,
  &__B,
  &__C {
    background: $on-primary;
    width: 0.7rem;
    height: 3rem;
    animation: spinner 1s infinite ease-in-out;
  }

  &__A {
    animation-delay: -0.16s;
  }

  &__C {
    animation-delay: 0.16s;
  }
}

@keyframes spinner {
  0%,
  80%,
  100% {
    transform: scaleY(1);
  }
  40% {
    transform: scaleY(1.4);
  }
}
</style>
