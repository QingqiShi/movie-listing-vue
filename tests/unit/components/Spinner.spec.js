import { shallowMount } from "@vue/test-utils";
import Spinner from "@/components/Spinner.vue";

describe("Spinner.vue", () => {
  it("renders", () => {
    const wrapper = shallowMount(Spinner);
    expect(wrapper).toMatchSnapshot();
  });
});
