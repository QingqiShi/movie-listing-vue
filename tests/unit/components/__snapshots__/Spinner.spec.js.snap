// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner.vue renders 1`] = `
<div class="spinner">
  <div class="spinner__wrapper">
    <div class="spinner__A"></div>
    <div class="spinner__B"></div>
    <div class="spinner__C"></div>
  </div>
</div>
`;
